src/contract.ts

```diff
@@ -111,37 +114,45 @@ export class AbstractContract {
 
     let error = `VerifyError: ${bsi.errstr}`;
 
-    const lastStepfExec = steps[stepCounter - 1].fExec;
 
-    if(!lastStepfExec) {
-      stepCounter = AbstractContract.findLastfExec(steps, stepCounter);
-    }
 
     // some time there is no opcodes, such as when sourcemap flag is closeed. 
     if(opcodes) {
       const offset = unlockingScriptASM.trim().split(' ').length;
       // the complete script may have op_return and data, but compiled output does not have it. So we need to make sure the index is in boundary.
-      const pc = Math.min(stepCounter -  offset,  opcodes.length -1);
-  
-      if(!result && opcodes[pc]) {
-  
-        const opcode = opcodes[pc]; 
+
+      const lastStepIndex = AbstractContract.findLastfExec(steps, stepCounter);
+
+      if(this._dataPart) {
```

Comment:
Same as above，this._dataPart could be empty string.

Reply:
f6181c8e687b92b652820e621b113ab6837ec23c